### ISSUE TITLE
Discussion on Setting self.tokenizer.pad_token_id

### ISSUE THREAD

**Issue by nkkbr** [LLaVA-VL/LLaVA-NeXT#338]
https://github.com/LLaVA-VL/LLaVA-NeXT/blob/79ef45a6d8b89b92d7a8525f077c3a3a9894a87d/llava/train/train.py#L1261-L1263

When configuring `attention_mask`, we typically do so as follows:

```python
attention_mask = input_ids.ne(self.tokenizer.pad_token_id)
```

However, if we set `eos_token_id` to be the same as `pad_token_id`, the `eos_token` will also be masked. This prevents the model from learning when to correctly stop generating text.

One possible solution is to set `self.tokenizer.pad_token_id` to 0. While this works to some extent, it’s not ideal because 0 is often a valid `token_id`. For example, in the Qwen-2.5-7B model, 0 corresponds to `"!"`. This workaround has some effect since `"!"` frequently appears at the end of sentences, and it’s relatively uncommon otherwise.

Wouldn't the best approach be to assign a dedicated `pad` token with a unique ID that does not overlap with any other token, including 0 and `eos_token_id`?
